core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -50,8 +51,30 @@ import io.opencensus.trace.{Span, Status}
  */
 //noinspection TypeAnnotation
 object WorkspaceService {
-  def constructor(dataSource: SlickDataSource, methodRepoDAO: MethodRepoDAO, cromiamDAO: ExecutionServiceDAO, executionServiceCluster: ExecutionServiceCluster, execServiceBatchSize: Int, methodConfigResolver: MethodConfigResolver, gcsDAO: GoogleServicesDAO, samDAO: SamDAO, notificationDAO: NotificationDAO, userServiceConstructor: UserInfo => UserService, genomicsServiceConstructor: UserInfo => GenomicsService, maxActiveWorkflowsTotal: Int, maxActiveWorkflowsPerUser: Int, workbenchMetricBaseName: String, submissionCostService: SubmissionCostService, config: WorkspaceServiceConfig)(userInfo: UserInfo)(implicit executionContext: ExecutionContext) = {
-    new WorkspaceService(userInfo, dataSource, methodRepoDAO, cromiamDAO, executionServiceCluster, execServiceBatchSize, methodConfigResolver, gcsDAO, samDAO, notificationDAO, userServiceConstructor, genomicsServiceConstructor, maxActiveWorkflowsTotal, maxActiveWorkflowsPerUser, workbenchMetricBaseName, submissionCostService, config)
+  def constructor(dataSource: SlickDataSource, methodRepoDAO: MethodRepoDAO, cromiamDAO: ExecutionServiceDAO,
+                  executionServiceCluster: ExecutionServiceCluster, execServiceBatchSize: Int,
+                  methodConfigResolver: MethodConfigResolver, gcsDAO: GoogleServicesDAO, samDAO: SamDAO,
+                  notificationDAO: NotificationDAO, userServiceConstructor: UserInfo => UserService,
+                  genomicsServiceConstructor: UserInfo => GenomicsService, maxActiveWorkflowsTotal: Int,
+                  maxActiveWorkflowsPerUser: Int, workbenchMetricBaseName: String, submissionCostService: SubmissionCostService,
+                  config: WorkspaceServiceConfig)
+                 (userInfo: UserInfo)
+                 (implicit executionContext: ExecutionContext) = {
+
+    // create the EntityManager along with its associated provider-builders. Since entities are only accessed
+    // in the context of a workspace, this is safe/correct to do here. We also want to use the same dataSource
+    // and execution context for the rawls entity provider that the workspace service uses.
+    val defaultEntityProviderBuilder = new RawlsEntityProviderBuilder(dataSource) // implicit executionContext
+    // soon: create a provider-builder for data repo and add it to the entity manager
+    val entityManager = new EntityManager(Set(defaultEntityProviderBuilder))
+
+    new WorkspaceService(userInfo, dataSource, methodRepoDAO, cromiamDAO,
+      executionServiceCluster, execServiceBatchSize,
+      methodConfigResolver, gcsDAO, samDAO,
+      notificationDAO, userServiceConstructor,
+      genomicsServiceConstructor, maxActiveWorkflowsTotal,
+      maxActiveWorkflowsPerUser, workbenchMetricBaseName, submissionCostService,
+      config, entityManager)
```

Comment:
a lot of the changes in this block are just line-wrapping.

Reply:
lol the constructor was 655 characters before